### PR TITLE
Add test for generated codes

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

To avoid problem like #1095 , we need to add some tests to the generated code. These tests could only avoid panic but don't check the logic.

### CODE REVIEW

File: cmd/chaos-builder/main.go [hunk 1]
```diff
@@ -48,20 +48,29 @@ const codeHeader = `// Copyright 2020 Chaos Mesh Authors.
...
 import (
 	"reflect"
 	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
-
 `
+	generatedTest := codeHeader + `
+import (
```
Comment (cwen0) [inline cmd/chaos-builder/main.go:63]: Can we define these imports as a const and avoid to define these in main function? such as `const testImport = `xxxx`

File: cmd/chaos-builder/main.go [hunk 2]
```diff
@@ -131,5 +141,12 @@ func init() {
 	}
 	fmt.Fprint(file, generatedCode)
 
+	generatedTest += testInit
+	file, err = os.Create("./api/v1alpha1/zz_generated.chaosmesh_test.go")
+	if err != nil {
+		log.Error(err, "fail to create file")
```
Comment (WangXiangUSTC) [inline cmd/chaos-builder/main.go:147]: how about exit with 1?

File: cmd/chaos-builder/test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+const testInit = `
+func init() {
+	faker.AddProvider("ioMethods", func(v reflect.Value) (interface{}, error) {
+		return []IoMethod{LookUp}, nil
+	})
+}
+`
+
+const testTemplate = `
```
Comment (WangXiangUSTC) [inline cmd/chaos-builder/test.go:29]: how about defining it after `testInit`?

Comment (YangKeao) [reply cmd/chaos-builder/test.go:29]: Sorry, I don't understand this comment.

Comment (WangXiangUSTC) [reply cmd/chaos-builder/test.go:40]: 🤕 just ignore it

File: cmd/chaos-builder/main.go (hunk 2 above)
Comment (YangKeao) [reply cmd/chaos-builder/main.go:133]: Fixed.

File: cmd/chaos-builder/main.go (hunk 1 above)
Comment (YangKeao) [reply cmd/chaos-builder/main.go]: Fixed

**Review by WangXiangUSTC — DISMISSED** [review]
LGTM

**Review by cwen0 — APPROVED** [review]
LGTM

**Review by cwen0 — APPROVED** [review]

**Review by g1eny0ung — APPROVED** [review]

**Pull request merged**